src/compute-client/src/controller/replicated.rs

```diff
@@ -138,11 +138,27 @@ where
     }
 }
 
-/// Additional information to store with pening peeks.
+/// Additional information to store with pending peeks.
 #[derive(Debug)]
 struct PendingPeek {
+    /// Replicas that have yet to respond to this peek.
+    unfinished: BTreeSet<ReplicaId>,
     /// The OpenTelemetry context for this peek.
-    otel_ctx: OpenTelemetryContext,
+    ///
+    /// This value is `Some` as long as we have not yet passed a response up the chain, and `None`
+    /// afterwards.
+    otel_ctx: Option<OpenTelemetryContext>,
```

Comment:
Hm, `otel_ctx` sound like something that is not functionally important, but just for tracing/logging to me

Reply:
It is just for tracing. In the beginning I thought we always need it when we construct a response, so this would work nicely to prevent us from producing duplicate responses. But this is actually only the case for cancel responses. For non-cancel responses, we use the otel context from the response we received, so we don't actually use the one stored here.

I thought about adding a separate field `served: bool`, but it seemed more of a hassle this way, because you'd need to remember to check and set an additional field when handling a peek response. But I'm fine with making that change if you feel strongly.

Reply:
Actually [line 392](https://github.com/MaterializeInc/materialize/pull/15504/files/3fce1170b725651f5274b33721cb56e0bbe09855#diff-3db47df6fd026e8e22e89e37f89c9409552803ba2112c55d881db1b32efb6e39R392) is the one where I thought that the contents of `otel_ctx` determine something functional. Maybe we could just rename it (to like `not_served` as in Some is interpreted as true, and None as false)?  But I'm fine either way :+1: